### PR TITLE
Multiclient that fits ethclient.Client

### PR DESCRIPTION
**Pull request by tomasmik**
Closes: https://github.com/mysteriumnetwork/node/issues/3568

### CODE REVIEW

File: client/bc_test.go [hunk 1]
```diff
@@ -0,0 +1,1572 @@
...
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+// Ensure, that EtherClientMock does implement EtherClient.
+// If this is not the case, regenerate this file with moq.
+var _ EtherClient = &EtherClientMock{}
+
+// EtherClientMock is a mock implementation of EtherClient.
```
Comment (vkuznecovas) [inline client/bc_test.go:20]: redundant comments?

File: client/ethmulticlient.go [hunk 2]
```diff
@@ -0,0 +1,697 @@
...
+		timeout:    defaulTimeout,
+		clients:    clients,
+		notifyDown: nil,
+	}, nil
+}
+
+// NewEthMultiClientNotifyDown creates a new multi clients eth client.
+//
+// Channel `notifications` must be given and will be used to push notifications to the
+// client if any nodes go down. The channel is closed when all the
```
Comment (vkuznecovas) [inline client/ethmulticlient.go:68]: incomplete comment.

File: client/ethmulticlient.go [hunk 3]
```diff
@@ -0,0 +1,697 @@
...
+
+func (c *EthMultiClient) closeNotify() {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	if c.notifyDown != nil {
+		close(c.notifyDown)
+		// Set equal to `nil` so that any hanging queries dont
+		// get broadcasted as failure on client close.
+		c.notifyDown = nil
```
Comment (vkuznecovas) [inline client/ethmulticlient.go:656]: i can see indefinite locks happening here. 

Scenario:

call closeNotify() and notifyDowntime() in multiple goroutines.

You'll run into a situation where c.notifyDown will beome nil after passing the c.notifyDown == nil check, meaning that the following select will block forever.


This needs to be addressed.

**Review by vkuznecovas — CHANGES_REQUESTED** [review]

File: client/bc_test.go (hunk 1 above)
Comment (tomasmik) [reply client/bc_test.go:20]: these are auto generated. I didn't really read them 🤷🏻

File: client/ethmulticlient.go (hunk 3 above)
Comment (tomasmik) [reply client/ethmulticlient.go:656]: nice catch

**Review by vkuznecovas — APPROVED** [review]

**Pull request merged**